apps/api/src/app/shared/helpers/content.service.ts

```diff
@@ -17,27 +18,16 @@ export class ContentService {
   extractVariables(content: string): string[] {
     if (!content) return [];
 
-    const regExp = /{{([a-zA-Z_][a-zA-Z0-9_-]*?)}}/gm;
-    const matchedItems = content.match(regExp);
+    const ast: hbs.AST.Program = Handlebars.parseWithoutProcessing(content);
 
-    const result = [];
-    if (!matchedItems || !Array.isArray(matchedItems)) {
-      return result;
-    }
-
-    for (const item of matchedItems) {
-      result.push(item.replace('{{', '').replace('}}', ''));
-    }
-
-    return result;
+    return this.getHandlebarVariables(ast.body);
```

Comment:
@ainouzgali could we add more usecases to the spec file here that covers what this functionality adds?